doc/source/run_engine_api.rst

```diff
@@ -0,0 +1,56 @@
+.. currentmodule:: bluesky.run_engine
+
+RunEngine API Documentation
+===========================
+
+.. autoclass:: RunEngine
+
+    .. automethod:: __call__
+
+    The RunEngine maintains a callback registry of functions that receive any
+    :doc:`documents` generated by plan execution. These methods add and remove
+    functions from that registry.
+
+    .. automethod:: subscribe
```

Comment:
I gave it a try (in a currently stashed local commit) and I don't like the results as much. I do like autosummary for something like our table built-in plans, where the functions are fairly self-describing and its docstring's top-line sentence is enough to explain its purpose. But for many RunEngine methods, the purpose requires more explanation. It's nice to put more detail up front.